datalad/distribution/publish.py

```diff
@@ -85,115 +86,230 @@ class Publish(Interface):
-
-        for handle_repo in handles_to_publish:
-
-            handle_name = handle_repo.path[len(
-                commonprefix([top_repo.path, handle_repo.path])):].strip("/")
-            set_upstream = False
-
-            if remote is not None and remote not in handle_repo.git_get_remotes():
-                if not remote_url:
-                    raise ValueError("No remote '%s' found. Provide REMOTE-URL"
-                                     " to add it." % remote)
-                lgr.info("Remote '%s' doesn't exist yet.")
-
-                # Fill in URL-Template:
-                remote_url = remote_url.replace("%%NAME",
-                                                handle_name.replace("/", "-"))
-                # Add remote
-                handle_repo.git_remote_add(remote, remote_url)
-                if remote_url_push:
-                    # Fill in template:
-                    remote_url_push = \
-                        remote_url_push.replace("%%NAME",
-                                                handle_name.replace("/", "-"))
-                    # Modify push url:
-                    handle_repo._git_custom_command('',
-                                                    ["git", "remote",
-                                                     "set-url",
-                                                     "--push", remote,
-                                                     remote_url_push])
-
-                lgr.info("Added remote '%s':\n %s (pull)\n%s (push)." %
-                         (remote, remote_url,
-                          remote_url_push if remote_url_push else remote_url))
+                 dest_pushurl=None, with_data=None, recursive=False):
+        # shortcut
+        ds = dataset
 
-            else:
-                # known remote: parameters remote-url-* currently invalid.
-                # This may change to adapt the existing remote.
-                if remote_url:
-                    lgr.warning("Remote '%s' already exists for handle '%s'. "
-                                "Ignoring remote-url %s." %
-                                (remote, handle_name, remote_url))
-                if remote_url_push:
-                    lgr.warning("Remote '%s' already exists for handle '%s'. "
-                                "Ignoring remote-url-push %s." %
-                                (remote, handle_name, remote_url_push))
-
-            # upstream branch needed for update (merge) and subsequent push,
-            # in case there is no.
+        if ds is not None and not isinstance(ds, Dataset):
+            ds = Dataset(ds)
+        if path is None:
+            if ds is None:
+                # no dataset, no component to publish, nothing to do
+                raise ValueError(
+                    "insufficient information for publication (needs at "
+                    "least a dataset or a path")
+        elif isinstance(path, list):
```

Comment:
If no `path` is given on the cmdline, it will come in as `[]`, enter this branch and do nothing, because it doesn't iterate over an empty list.